### PR TITLE
hdtorrents: add banner and code cleanup

### PR DESCRIPTION
**Pull request by ngosang**
Tested

### CODE REVIEW

File: src/Jackett.Common/Indexers/HDTorrents.cs [hunk 1]
```diff
@@ -20,6 +21,16 @@ public class HDTorrents : BaseWebIndexer
     {
         private string SearchUrl => SiteLink + "torrents.php?";
         private string LoginUrl => SiteLink + "login.php";
+        private readonly Regex BannerRegex = new Regex(@"src=\\'./([^']+)\\'", RegexOptions.IgnoreCase);
+        private readonly HashSet<string> FreeleechRanks = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/HDTorrents.cs:25]: Private variable naming. Update names elsewhere too.

```suggestion
        private readonly Regex _bannerRegex = new Regex(@"src=\\'./([^']+)\\'", RegexOptions.IgnoreCase);
        private readonly HashSet<string> _freeleechRanks = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

File: src/Jackett.Common/Indexers/HDTorrents.cs [hunk 2]
```diff
@@ -165,51 +173,58 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    var dateString = string.Join(" ", dateTag.Attributes.Select(attr => attr.Name));
-                    release.PublishDate = DateTime.ParseExact(dateString, "dd MMM yyyy HH:mm:ss zz00", CultureInfo.InvariantCulture).ToLocalTime();
-
-                    var category = row.FirstElementChild.FirstElementChild.GetAttribute("href").Split('=')[1];
-                    release.Category = MapTrackerCatToNewznab(category);
-
-                    release.UploadVolumeFactor = 1;
+                    long? grabs = null;
+                    if (ParseUtil.TryCoerceLong(row.Children[endIndex - 1].TextContent, out var rGrabs))
+                        grabs = rGrabs;
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/HDTorrents.cs:187]:
```suggestion

                    var grabs = ParseUtil.TryCoerceLong(row.Children[endIndex - 1].TextContent, out var rGrabs)
                        ? (long?)rGrabs
                        : null;
```

File: src/Jackett.Common/Indexers/HDTorrents.cs [hunk 3]
```diff
@@ -29,26 +40,23 @@ public class HDTorrents : BaseWebIndexer
...
-                logger: l,
-                p: ps,
-                configData: new ConfigurationDataBasicLogin())
+            : base("HD-Torrents",
+                   description: "HD-Torrents is a private torrent website with HD torrents and strict rules on their content.",
+                   link: "https://hdts.ru/", // Domain https://hdts.ru/ seems more reliable
+                   caps: new TorznabCapabilities
+                   {
+                       SupportsImdbMovieSearch = true,
+                       SupportsImdbTVSearch = true
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/HDTorrents.cs:52]: You recently commented out this line from all other trackers, we should do the same here I think.

Comment (ngosang) [reply src/Jackett.Common/Indexers/HDTorrents.cs]: rebased

**Comment by garfield69** [comment]
Jackett v0.14.541

**Pull request merged**